go/src/koding/kites/metrics/metrics.go

```diff
@@ -42,9 +42,14 @@ func NewWithPath(boltPath, app string) (*Metrics, error) {
 // NewWithDB creates a new metric collector with given bolt db instance. See New
 // func.
 func NewWithDB(db *bolt.DB, app string) (*Metrics, error) {
-	storage, err := NewBoltStorageWithDB(db)
+	var storage Storage
+	var err error
+	if db != nil {
+		storage, err = NewBoltStorageWithDB(db)
```

Comment:
We have that losing issue on [here](https://github.com/koding/koding/pull/11218#discussion_r118312089) as well. That is in my list to handle them gracefully.